### ISSUE TITLE
No menu items available in Selection menu

### ISSUE THREAD

**Issue by lmcbout** [theia-ide/theia#731]
Menu "Selection" available, but all options are greyed out. Should we have this menu option available?
![peek 2017-10-23 selectionmenu](https://user-images.githubusercontent.com/419051/31948182-3b2e1216-b8a4-11e7-80dd-e6cbb0cef71d.gif)

Some discussion took place for issue in yangster:
ref: https://github.com/theia-ide/yangster/issues/26
- Most of these don't make sense for the diagram. Which ones would you like to use?
I don't think that Theia can deactivate entire menus.

-If we use the diagram display and if none of the menu items apply to any diagram, then the menu "Selection" should not clutter the menu bar, it should be invisible when the user is editing/viewing a diagram

**Comment by kittaakos** [comment]
I would rephrase this: do we want to dynamically update (hide) a particular sub-menu from the main menu if none of the corresponding commands have an active handler?

I believe one has to also keep in mind whether we show an inactive menu item, or we simply omit it when no active handlers are available for the command.

Also related: #446.

**Comment by svenefftinge** [comment]
I think the behavior is fine, since removing and adding menus based on context would be bad (as we learned in Eclipse). 
I'd like to close this. Objections?

**Comment by kittaakos** [comment]
> I'd like to close this.

+1

**Comment by lmcbout** [comment]
Personally, I find it weird having a pull-down menu with no items selectable. Second, The selection menu is also completely empty when starting Theia, if you click on it, a few pixels shows the menu is selected, but  nothing else, no menus. Another example. When running yangster, menu items in the "Edit" sub-menu are not needed  ("Cut/COPY/PASTE") If there is no plan to make them available, why do we display them when the user is only handling some UI display?

**Comment by hexa00** [comment]
If we don't change the menus based on context we should fix the display when there's no items... like adding a empty menu item?

**Comment by svenefftinge** [comment]
The totally empty main menu (#751) is a bug.

**Comment by svenefftinge** [comment]
>  When running yangster, menu items in the "Edit" sub-menu are not needed ("Cut/COPY/PASTE") If there is no plan to make them available, why do we display them when the user is only handling some UI display?

I don't understand, could you explain what you mean?

**Comment by lmcbout** [comment]
If you look at the yangster issue: 
https://github.com/theia-ide/yangster/issues/29
There seems to be no plan to  edit directly into the diagram. Under the "Edit" menu, there are some options to modify the editor : Copy/CUT/PASTE. If there is no plan to use those features, why are they there !!

**Comment by svenefftinge** [comment]
Copy/Cut/Paste is supported in yangster, generally, just not in the diagram.

**Comment by svenefftinge** [comment]
Can this ticket be closed, or is there anything we should take action on?
